utils/utils.py

```diff
@@ -176,6 +228,136 @@ def options_for_gather():
+        "where possible.",
+    ]))
+    parser.add_argument("--debug", action="store_true",
+                        help="Print out more stuff. Useful with '--serial'")
+    parser.add_argument("--lambda", action="store_true", help="".join([
+        "Run certain scanners inside Amazon Lambda instead of locally.",
+    ]))
+    parser.add_argument("--lambda-profile", nargs=1, help="".join([
+        "When running Lambda-related commands, use a specified AWS named ",
+        "profile. Credentials/config for this named profile should already ",
+        "be configured separately in the execution environment.",
+    ]))
+    parser.add_argument("--meta", action="store_true", help="".join([
+        "Append some additional columns to each row with information about ",
+        "the scan itself. This includes start/end times and durations, as ",
+        "well as any encountered errors. When also using '--lambda', ",
+        "additional, Lambda-specific information will be appended.",
+    ]))
+    parser.add_argument("--scan", nargs=1, required=True,
+                        help="Comma-separated list of scanners (required).")
+    parser.add_argument("--sort", action="store_true", help="".join([
+        "Sort result CSVs by domain name, alphabetically. (Note: this causes ",
+        "the entire dataset to be read into memory.)",
+    ]))
+    parser.add_argument("--serial", action="store_true", help="".join([
+        "Disable parallelization, force each task to be done simultaneously. ",
+        "Helpful for testing and debugging.",
+    ]))
+
+    parser.add_argument("--suffix", nargs=1, help="".join([
+        "Add a suffix to all input domains. For example, a --suffix of ",
+        "'virginia.gov' will add '.virginia.gov' to the end of all ",
+        "input domains."
+    ]))
+    parser.add_argument("--output", nargs=1, default=["./"], help="".join([
+        "Where to output the 'cache/' and 'results/' directories. ",
+        "Defaults to './'.",
+    ]))
+    parser.add_argument("--workers", nargs=1,
+                        help="Limit parallel threads per-scanner to a number.")
+    # TODO: Should workers have a default value?
+
+    return parser
+
+
+def options_for_scan():
+    """
+    Parse options for the ``scan`` command.
+
+    :rtype: dict
+
+    Impure
+        Reads from sys.argv.
+
+    """
+    parser = build_scan_options_parser()
+    parsed = parser.parse_args()
+
+    opts = parsed.__dict__
+    opts = {k: opts[k] for k in opts if opts[k] is not None}
+
+    if opts.get("lambda_profile") and not opts.get("lambda"):
+            raise argparse.ArgumentTypeError(
+                "Can't set lambda profile unless lambda flag is set.")
+
+    # We know we want one value, but the ``nargs`` flag means we get a list.
+    should_be_singles = [
+        "lambda_profile",
+        "output",
+        "scan",
+        "suffix",
+        "workers",
+    ]
```

Comment:
This is a lot of custom logic controlling how arguments are parsed. I would rather change the flags to make it easier to parse them consistently, then to create a sophisticated parser that molds itself around how they are currently used. 

I understand why the dynamic gatherer names need some special code, but not why the "singles" need their own code. Is there something we should look at changing in the CLI interface to make argument parsing easier?

Reply:
I could be missing something, but I don't think that argparse lets us get away from this particular quirk. We use `nargs` to tell it that we expect some number of arguments (usually 1) to that flag, and it returns a list from any use of `nargs` even if the value is `1`. The other option here was changing code further down the chain to expect lists rather than single values, but I felt it was better to address it in a single spot.

Reply:
I'm happy to simplify the code by changing the CLI as it is; I had been trying to avoid that while working on this PR.